lib/spack/spack/package.py

```diff
@@ -326,8 +326,403 @@ def remove_files_from_view(self, view, merge_map):
+            if specs:
+                spack.store.db.remove(specs[0])
+                tty.msg("Removed stale DB entry for %s" % spec.short_spec)
+                return
+            else:
+                raise InstallError(str(spec) + " is not installed.")
+
+        if not force:
+            dependents = spack.store.db.installed_relatives(
+                spec, 'parents', True)
+            if dependents:
+                raise PackageStillNeededError(spec, dependents)
+
+        # Try to get the pcakage for the spec
+        try:
+            pkg = spec.package
+        except spack.repo.UnknownEntityError:
+            pkg = None
+
+        # Pre-uninstall hook runs first.
+        with spack.store.db.prefix_write_lock(spec):
+
+            if pkg is not None:
+                spack.hooks.pre_uninstall(spec)
+
+            # Uninstalling in Spack only requires removing the prefix.
+            if not spec.external:
+                msg = 'Deleting package prefix [{0}]'
+                tty.debug(msg.format(spec.short_spec))
+                spack.store.layout.remove_install_directory(spec)
+            # Delete DB entry
+            msg = 'Deleting DB entry [{0}]'
+            tty.debug(msg.format(spec.short_spec))
+            spack.store.db.remove(spec)
+
+        if pkg is not None:
+            spack.hooks.post_uninstall(spec)
+
+        tty.msg("Successfully uninstalled %s" % spec.short_spec)
+
+    def do_uninstall(self, force=False):
+        """Uninstall this package by spec."""
+        # delegate to instance-less method.
+        PackageCore.uninstall_by_spec(self.spec, force)
+
+    def format_doc(self, **kwargs):
+        """Wrap doc string at 72 characters and format nicely"""
+        indent = kwargs.get('indent', 0)
+
+        if not self.__doc__:
+            return ""
+
+        doc = re.sub(r'\s+', ' ', self.__doc__)
+        lines = textwrap.wrap(doc, 72)
+        results = StringIO()
+        for line in lines:
+            results.write((" " * indent) + line + "\n")
+        return results.getvalue()
+
+    @property
+    def rpath(self):
+        """Get the rpath this package links with, as a list of paths."""
+        rpaths = [self.prefix.lib, self.prefix.lib64]
+        deps = self.spec.dependencies(deptype='link')
+        rpaths.extend(d.prefix.lib for d in deps
+                      if os.path.isdir(d.prefix.lib))
+        rpaths.extend(d.prefix.lib64 for d in deps
+                      if os.path.isdir(d.prefix.lib64))
+        return rpaths
+
+    @property
+    def rpath_args(self):
+        """
+        Get the rpath args as a string, with -Wl,-rpath, for each element
+        """
+        return " ".join("-Wl,-rpath,%s" % p for p in self.rpath)
+
+
+class DependenciesPackage(PackageCore):
+    """Dependencies-only (aka Bundle, No-Source) installation package."""
+    #: The one and only phase
+    phases = ['install']
+
+
+class PackageBase(PackageCore):
```

Comment:
I agree though this would be a significant, backward-incompatible change.

This package name appears in 114 locations (per `git grep PackageBase`) affecting 24 separate files between the documentation and code.

If this kind of backward incompatibility is okay, then I'll make the change.

Reply:
One issue here: not all `SourcePackage` instances actually install source.  Some of them just download binaries.  We may also have packages that are just stubs for proprietary vendor-supplied Spack packages, where there would only be a binary available for end users.  So is there another name that covers those bases?